### PR TITLE
Added voice to RelatedAudioItem, added RelatedVideoItem

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

File: LDSContent/ItemPackage+RelatedAudioItem.swift [hunk 1]
```diff
@@ -31,15 +31,16 @@ public extension ItemPackage {
...
         static let subitemID = Expression<Int64>("subitem_id")
         static let mediaURL = Expression<String>("media_url")
-        static let fileSize = Expression<Int>("file_size")
+        static let fileSize = Expression<Int64>("file_size")
         static let duration = Expression<Int>("duration")
+        static let voice = Expression<RelatedAudioVoice?>("related_audio_voice_id")
         
         static func fromRow(row: Row) -> RelatedAudioItem {
-            return RelatedAudioItem(id: row[id], subitemID: row[subitemID], mediaURL: NSURL(string: row[mediaURL])!, fileSize: row[fileSize], duration: row[duration])
+            return RelatedAudioItem(id: row[id], subitemID: row[subitemID], mediaURL: NSURL(string: row[mediaURL])!, fileSize: row[fileSize], duration: row[duration], voice: row.get(voice))
```
Comment (nickmshelley) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: I know you didn't do this as part of this PR, but how did a `!` get in there?

File: LDSContent/ItemPackage+RelatedAudioItem.swift [hunk 2]
```diff
@@ -31,15 +31,16 @@ public extension ItemPackage {
...
         }
         
         static func fromNamespacedRow(row: Row) -> RelatedAudioItem {
-            return RelatedAudioItem(id: row[RelatedAudioItemTable.table[id]], subitemID: row[RelatedAudioItemTable.table[subitemID]], mediaURL: NSURL(string: row[RelatedAudioItemTable.table[mediaURL]])!, fileSize: row[RelatedAudioItemTable.table[fileSize]], duration: row[RelatedAudioItemTable.table[duration]])
+            return RelatedAudioItem(id: row[RelatedAudioItemTable.table[id]], subitemID: row[RelatedAudioItemTable.table[subitemID]], mediaURL: NSURL(string: row[RelatedAudioItemTable.table[mediaURL]])!, fileSize: row[RelatedAudioItemTable.table[fileSize]], duration: row[RelatedAudioItemTable.table[duration]], voice: row.get(RelatedAudioItemTable.voice))
```
Comment (nickmshelley) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: Same here, what's with the `!`?

File: LDSContent/ItemPackage+RelatedVideoItem.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+public extension ItemPackage {
+    
+    class RelatedVideoItemTable {
+        
+        static let table = Table("related_video_item")
+        static let id = Expression<Int64>("_id")
+        static let subitemID = Expression<Int64>("subitem_id")
+        static let posterURL = Expression<String>("poster_url")
+        static let videoID = Expression<String>("data_video_id")
+        static let title = Expression<String>("data_video_title")
```
Comment (nickmshelley) [inline LDSContent/ItemPackage+RelatedVideoItem.swift]: Why the `data_` prefixes here?

File: LDSContent/ItemPackage+RelatedAudioItem.swift (hunk 2 above)
Comment (stephanheilner) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: Following how @hiltonc did the mediaURL in RelatedAudioItem.

File: LDSContent/ItemPackage+RelatedAudioItem.swift (hunk 1 above)
Comment (stephanheilner) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: Following how @hiltonc did the mediaURL in RelatedAudioItem.

Comment (nickmshelley) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: I don't like them, but if you want to keep them that's fine.

**Comment by nickmshelley** [comment]
.

**Comment by hiltonc** [comment]
The more ! the happier!

**Comment by hiltonc** [comment]
I guess throwing or returning nil would also be an option.

File: LDSContent/ItemPackage+RelatedVideoItem.swift [hunk 4]
```diff
@@ -30,16 +30,16 @@ public extension ItemPackage {
...
-        static let posterURL = Expression<String>("poster_url")
-        static let videoID = Expression<String>("data_video_id")
-        static let title = Expression<String>("data_video_title")
+        static let posterURL = Expression<String?>("poster_url")
+        static let videoID = Expression<String>("video_id")
+        static let title = Expression<String>("title")
         
         static func fromRow(row: Row) -> RelatedVideoItem {
-            return RelatedVideoItem(id: row[id], subitemID: row[subitemID], posterURL: NSURL(string: row[posterURL])!, videoID: row[videoID], title: row[title])
+            return RelatedVideoItem(id: row[id], subitemID: row[subitemID], posterURL: NSURL(string: row[posterURL] ?? ""), videoID: row[videoID], title: row[title])
```
Comment (nickmshelley) [inline LDSContent/ItemPackage+RelatedVideoItem.swift]: `NSURL(string: "")` doesn't return `nil`, so if `row[posterURL]` isn't there the `posterURL` property will still have an empty `NSURL`, which I don't think is what we want.

Comment (hiltonc) [inline LDSContent/ItemPackage+RelatedVideoItem.swift]: row[posterURL].flatMap { NSURL(string: $0) }

File: LDSContent/ItemPackage+RelatedVideoItem.swift [hunk 5]
```diff
@@ -30,16 +30,16 @@ public extension ItemPackage {
...
         }
         
         static func fromNamespacedRow(row: Row) -> RelatedVideoItem {
-            return RelatedVideoItem(id: row[RelatedAudioItemTable.table[id]], subitemID: row[RelatedAudioItemTable.table[subitemID]], posterURL: NSURL(string: row[RelatedAudioItemTable.table[posterURL]])!, videoID: row[RelatedAudioItemTable.table[videoID]], title: row[RelatedAudioItemTable.table[title]])
+            return RelatedVideoItem(id: row[RelatedAudioItemTable.table[id]], subitemID: row[RelatedAudioItemTable.table[subitemID]], posterURL: NSURL(string: row[RelatedAudioItemTable.table[posterURL]] ?? ""), videoID: row[RelatedAudioItemTable.table[videoID]], title: row[RelatedAudioItemTable.table[title]])
```
Comment (nickmshelley) [inline LDSContent/ItemPackage+RelatedVideoItem.swift]: Same here.

**Comment by nickmshelley** [comment]
👍  Thanks for getting rid of the `!`s!

**Comment by stephanheilner** [comment]
The failures are missing column names. This will fail until content packager uses this code. So would it be okay to merge?

**Comment by hiltonc** [comment]
ok with me

**Pull request merged**